litestar/openapi/plugins.py

```diff
@@ -0,0 +1,687 @@
+                {body}
+            </html>
+        """.encode()
+
+
+class RedocRenderPlugin(OpenAPIRenderPlugin):
+    """Render an OpenAPI schema using Redoc."""
+
+    def __init__(
+        self,
+        *,
+        version: str = "next",
+        js_url: str | None = None,
+        google_fonts: bool = True,
+        path: str | Sequence[str] = "/redoc",
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the OpenAPI UI render plugin.
+
+        Args:
+            version: Redoc version to download from the CDN. If js_url is provided, this is ignored.
+            js_url: Download url for the Redoc JS bundle. If not provided, the version will be used to construct the url.
+            google_fonts: Download google fonts via CDN. Should be set to False when not using a CDN.
+            path: Path to serve the OpenAPI UI at.
+            **kwargs: Additional arguments to pass to the base class.
+        """
+        self.js_url = js_url or f"https://cdn.jsdelivr.net/npm/redoc@{version}/bundles/redoc.standalone.js"
+        self.google_fonts = google_fonts
+        super().__init__(path=path, **kwargs)
+
+    def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
+        """Render an HTML page for Redoc.
+
+        .. note:: override this method to customize the template.
+
+        Args:
+            request: The request.
+            openapi_schema: The OpenAPI schema as a dictionary.
+
+        Returns:
+            A rendered html string.
+        """
+
+        head = f"""
+          <head>
+            <title>{openapi_schema["info"]["title"]}</title>
+            {self.favicon}
+            <meta charset="utf-8"/>
+            <meta name="viewport" content="width=device-width, initial-scale=1">
+            """
+
+        if self.google_fonts:
+            head += """
+            <link href="https://fonts.googleapis.com/css?family=Montserrat:300,400,700|Roboto:300,400,700" rel="stylesheet">
+            """
+
+        head += f"""
+            <script src="{self.js_url}" crossorigin></script>
+            {self.style}
+          </head>
+        """
+
+        body = b"".join(
+            [
+                b"<body><div id='redoc-container'/><script type='text/javascript'>Redoc.init(",
+                self.render_json(request, openapi_schema),
+                b",undefined,document.getElementById('redoc-container'))</script></body>",
+            ]
+        )
+
+        return b"".join(
+            [
+                b"<!DOCTYPE html><html>",
+                head.encode(),
+                body,
+                b"</html>",
+            ]
+        )
+
+
+class ScalarRenderPlugin(OpenAPIRenderPlugin):
+    """Plugin to render an OpenAPI schema using Scalar.
+
+    .. versionadded:: 2.8.0
+    """
+
+    _default_litestar_css_url = "https://cdn.jsdelivr.net/gh/litestar-org/branding@main/assets/openapi/scalar.css"
+    _default_scalar_css_url = "/schema/_litestar-internal/scalar.css"
```

Comment:
I don't think the `/schema/` part of this can be hard coded as it configurable on `OpenAPIConfig`